File: services/portfolioService.js
const PORTFOLIO_DATA = {
    name: "Alish Shrestha",
    age: 18,
    location: "Changu Narayan-01, Bhaktapur, Nepal",
    email: "shresthaalish444@gmail.com",
    college: "Softwarica College of IT & E-Commerce (Coventry University, UK)",
    degree: "BSc (Hons) Computer Science with Artificial Intelligence",
    secondarySchool: "Khwopa Secondary School, Bhaktapur (Computer Science Specialization, 2023-2025)",
    primarySchool: "North East English Secondary School, Bhaktapur",
    skills: {
        programming: ["Python", "JavaScript", "HTML5", "CSS3", "C++", "C#"],
        ai_ml: ["TensorFlow", "PyTorch", "Machine Learning", "Neural Networks", "Deep Learning", "Data Science"],
        tools: ["VS Code", "GitHub", "Git", "Terminal", "Arduino"],
        creative: ["Photography", "Video Editing", "UI/UX Design", "Communication"]
    },
    projects: [
        {
            name: "Yatra Travel Agency",
            description: "Travel booking & tour guide website built for Nepal tourism.",
            link: "https://yatrala.netlify.app",
            github: "https://github.com/fwabyss0/Yatra.git"
        },
        {
            name: "Printing Resolution",
            description: "Online printing service platform for custom prints and designs.",
            link: "https://printresolution.netlify.app",
            github: "https://github.com/fwabyss0/pr"
        },
        {
            name: "Interactive Portfolio & AI Assistant (Abyss)",
            description: "Personal interactive web portfolio featuring real-time AI assistant.",
            github: "https://github.com/fwabyss0/Protfolio.git"
        }
    ],
    social: {
        github: "https://github.com/fwabyss0",
        linkedin: "https://www.linkedin.com/in/alish-shrestha-4276b8379/",
        instagram: "https://www.instagram.com/aliisshhhhhh/",
        facebook: "https://www.facebook.com/alish.shrestha.138982",
        discord: "fwabyss"
    },
    cv: "Alish_Shrestha_CV.pdf"
};

function isPortfolioQuery(message) {
    const msg = message.toLowerCase();
    const keywords = [
        'alish', 'who are you', 'your name', 'created you', 'creator', 'owner',
        'skills', 'programming', 'languages', 'tech stack', 'coding', 'experience',
        'education', 'college', 'university', 'softwarica', 'khwopa', 'school',
        'projects', 'yatra', 'printing resolution', 'portfolio', 'built', 'work',
        'contact', 'email', 'reach', 'social', 'github', 'linkedin', 'instagram', 'facebook', 'discord', 'fwabyss',
        'location', 'where', 'from', 'live', 'nepal', 'bhaktapur', 'age', 'how old', 'cv', 'resume'
    ];
    return keywords.some(kw => msg.includes(kw));
}

function getPortfolioContext() {
    return `
ABOUT ALISH SHRESTHA (Portfolio Owner):
- Name: Alish Shrestha
- Age: 18 years old
- Location: Changu Narayan-01, Bhaktapur, Nepal 🇳🇵
- Email: shresthaalish444@gmail.com
- Status: Passionate AI student, developer, and creative designer

EDUCATION:
- Currently: Softwarica College of IT & E-Commerce (Coventry University, UK) - Bachelor's in Artificial Intelligence (BSc Hons)
- 2023-2025: Khwopa Secondary School, Dekocha-06, Bhaktapur - Computer Science Specialization
- Primary: North East English Secondary School, Changunarayan-03, Bhaktapur

SKILLS:
- Programming: Python, JavaScript, HTML5/CSS3, C++, C#
- AI / ML: TensorFlow, PyTorch, Machine Learning, Neural Networks, Deep Learning, Data Science
- Tools & Platforms: VS Code, GitHub, Git, Terminal, Arduino
- Creative: Photography, Video Editing, UI/UX Design, Communication

PROJECTS:
1. Yatra Travel Agency - Travel booking website for Nepal (https://yatrala.netlify.app | Repo: https://github.com/fwabyss0/Yatra.git)
2. Printing Resolution - Online printing service (https://printresolution.netlify.app | Repo: https://github.com/fwabyss0/pr)
3. Interactive Portfolio with AI Assistant (Abyss) - Personal portfolio (https://github.com/fwabyss0/Protfolio.git)

SOCIAL & CONTACT LINKS:
- GitHub: https://github.com/fwabyss0
- LinkedIn: https://www.linkedin.com/in/alish-shrestha-4276b8379/
- Instagram: @aliisshhhhhh (https://www.instagram.com/aliisshhhhhh/)
- Facebook: https://www.facebook.com/alish.shrestha.138982
- Discord: fwabyss
- Download CV: Alish_Shrestha_CV.pdf

PORTFOLIO RULE: Always prioritize this data when asked about Alish, his skills, projects, education, or contact details. Keep replies clear and engaging.
`;
}

function getPortfolioDirectResponse(message) {
    const msg = message.toLowerCase();

    if (msg.includes('discord') || msg.includes('fwabyss')) {
        return "You can add Alish on Discord with username: **fwabyss**! Feel free to send a friend request for tech chats or gaming. 🎮";
    }
    if (msg.includes('email') || msg.includes('contact') || msg.includes('reach')) {
        return "You can reach Alish directly at: [shresthaalish444@gmail.com](mailto:shresthaalish444@gmail.com) 📧";
    }
    if (msg.includes('cv') || msg.includes('resume')) {
        return "You can download Alish's official CV here: [Download CV](Alish_Shrestha_CV.pdf) 📄";
    }
    if (msg.includes('github')) {
        return "Check out Alish's code repositories on GitHub: [github.com/fwabyss0](https://github.com/fwabyss0) 💻";
    }
    if (msg.includes('linkedin')) {
        return "Connect with Alish on LinkedIn: [Alish Shrestha LinkedIn](https://www.linkedin.com/in/alish-shrestha-4276b8379/) 💼";
    }
    if (msg.includes('instagram')) {
        return "Follow Alish on Instagram: [@aliisshhhhhh](https://www.instagram.com/aliisshhhhhh/) 📸";
    }
    if (msg.includes('facebook')) {
        return "Connect with Alish on Facebook: [Alish Shrestha Facebook](https://www.facebook.com/alish.shrestha.138982) 👤";
    }

    return null;
}

module.exports = {
    PORTFOLIO_DATA,
    isPortfolioQuery,
    getPortfolioContext,
    getPortfolioDirectResponse
};

File: backend/services/portfolio_service.py
"""
portfolio_service.py
Handles all portfolio-related queries locally (no AI call needed).
"""
import random

PORTFOLIO_DATA = {
    "name": "Alish Shrestha",
    "age": 18,
    "location": "Changu Narayan-01, Bhaktapur, Nepal",
    "email": "shresthaalish444@gmail.com",
    "college": "Softwarica College of IT & E-Commerce (Coventry University, UK)",
    "degree": "BSc (Hons) Computer Science with Artificial Intelligence",
    "secondary_school": "Khwopa Secondary School, Bhaktapur (Computer Science, 2023-2025)",
    "primary_school": "North East English Secondary School, Bhaktapur",
    "skills": {
        "programming": ["Python", "JavaScript", "HTML5", "CSS3", "C++", "C#"],
        "ai_ml": ["TensorFlow", "PyTorch", "Machine Learning", "Neural Networks", "Deep Learning", "Data Science"],
        "tools": ["VS Code", "GitHub", "Git", "Terminal", "Arduino"],
        "creative": ["Photography", "Video Editing", "UI/UX Design", "Communication"]
    },
    "projects": [
        {
            "name": "Yatra Travel Agency",
            "description": "Travel booking & tour guide website built for Nepal tourism.",
            "link": "https://yatrala.netlify.app",
            "github": "https://github.com/fwabyss0/Yatra.git"
        },
        {
            "name": "Printing Resolution",
            "description": "Online printing service platform for custom prints and designs.",
            "link": "https://printresolution.netlify.app",
            "github": "https://github.com/fwabyss0/pr"
        },
        {
            "name": "Interactive Portfolio & AI Assistant (Abyss)",
            "description": "Personal interactive web portfolio featuring real-time AI assistant.",
            "github": "https://github.com/fwabyss0/Protfolio.git"
        }
    ],
    "social": {
        "github": "https://github.com/fwabyss0",
        "linkedin": "https://www.linkedin.com/in/alish-shrestha-4276b8379/",
        "instagram": "https://www.instagram.com/aliisshhhhhh/",
        "facebook": "https://www.facebook.com/alish.shrestha.138982",
        "discord": "fwabyss"
    },
    "cv": "Alish_Shrestha_CV.pdf"
}

PORTFOLIO_KEYWORDS = {
    "greeting": ["hi", "hello", "hey", "namaste", "good morning", "good afternoon", "good evening", "sup", "yo"],
    "about": ["who is alish", "about alish", "tell me about", "introduce", "background", "who are you", "alish"],
    "age": ["age", "how old", "years old", "born", "birthday"],
    "skills": ["skills", "programming", "tech stack", "coding", "abilities", "technologies", "expertise", "languages"],
    "education": ["education", "study", "college", "university", "softwarica", "coventry", "school", "degree", "learning"],
    "projects": ["projects", "work", "portfolio", "built", "created", "developed", "yatra", "printing resolution"],
    "contact": ["contact", "email", "reach", "get in touch"],
    "cv": ["cv", "resume", "curriculum vitae", "download cv"],
    "location": ["location", "where", "from", "live", "nepal", "bhaktapur", "address"],
    "social_github": ["github", "git", "code", "repositories", "repos"],
    "social_linkedin": ["linkedin", "professional", "network", "career"],
    "social_instagram": ["instagram", "insta", "ig", "photos"],
    "social_facebook": ["facebook", "fb", "friends"],
    "social_discord": ["discord", "gaming", "fwabyss", "friend request"],
    "abyss": ["abyss", "chatbot", "ai assistant", "who are you", "what are you"],
}


def is_portfolio_query(message: str) -> bool:
    msg = message.lower()
    for keywords in PORTFOLIO_KEYWORDS.values():
        if any(kw in msg for kw in keywords):
            return True
    return False


def get_portfolio_response(message: str) -> str | None:
    msg = message.lower()
    d = PORTFOLIO_DATA

    # Greetings
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["greeting"]):
        return random.choice([
            "Hello! 👋 I'm Abyss, Alish's AI assistant! How can I help you today?",
            "Hi there! 😊 I'm Abyss! What can I tell you about Alish?",
            "Hey! 🎉 Abyss here — ask me anything about Alish or anything else!",
            "Namaste! 🙏 I'm Abyss. How can I help you?"
        ])

    # Abyss identity
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["abyss"]) and "alish" not in msg:
        return (
            "I'm **Abyss** 🤖 — Alish Shrestha's personal AI assistant! "
            "I can answer questions about Alish, help with general knowledge, check live weather, "
            "fetch crypto prices, solve math problems, and much more. What would you like to know?"
        )

    # Social media — Discord
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["social_discord"]):
        return f"🎮 Add Alish on Discord: **{d['social']['discord']}** — send him a friend request to chat!"

    # Social media — GitHub
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["social_github"]):
        return (
            f"👨‍💻 Check out Alish's code on GitHub:\n\n"
            f"[github.com/fwabyss0]({d['social']['github']})"
        )

    # Social media — LinkedIn
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["social_linkedin"]):
        return (
            f"💼 Connect with Alish on LinkedIn:\n\n"
            f"[Alish Shrestha LinkedIn]({d['social']['linkedin']})"
        )

    # Social media — Instagram
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["social_instagram"]):
        return (
            f"📸 Follow Alish on Instagram:\n\n"
            f"[@aliisshhhhhh]({d['social']['instagram']})"
        )

    # Social media — Facebook
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["social_facebook"]):
        return (
            f"👥 Connect with Alish on Facebook:\n\n"
            f"[Alish Shrestha]({d['social']['facebook']})"
        )

    # CV / Resume
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["cv"]):
        return f"📄 You can download Alish's official CV here: [{d['cv']}]({d['cv']})"

    # Contact / Email
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["contact"]):
        return (
            f"📧 You can reach Alish directly at: [{d['email']}](mailto:{d['email']})\n\n"
            f"Or connect via LinkedIn, GitHub, or Instagram!"
        )

    # Location
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["location"]):
        return f"📍 Alish is from **{d['location']}** — a beautiful UNESCO heritage city in Nepal! 🇳🇵"

    # Age
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["age"]):
        return f"🎂 Alish is **{d['age']} years old** — young and full of ambition!"

    # Education
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["education"]):
        return (
            f"🎓 **Alish's Education:**\n\n"
            f"- **Currently:** {d['college']} — {d['degree']}\n"
            f"- **Secondary:** {d['secondary_school']}\n"
            f"- **Primary:** {d['primary_school']}"
        )

    # Skills
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["skills"]):
        skills = d["skills"]
        return (
            f"💻 **Alish's Skills:**\n\n"
            f"- **Programming:** {', '.join(skills['programming'])}\n"
            f"- **AI / ML:** {', '.join(skills['ai_ml'])}\n"
            f"- **Tools:** {', '.join(skills['tools'])}\n"
            f"- **Creative:** {', '.join(skills['creative'])}"
        )

    # Projects
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["projects"]):
        lines = ["🚀 **Alish's Projects:**\n"]
        for i, p in enumerate(d["projects"], 1):
            line = f"{i}. **{p['name']}** — {p['description']}"
            if p.get("link"):
                line += f" | [Live]({p['link']})"
            if p.get("github"):
                line += f" | [GitHub]({p['github']})"
            lines.append(line)
        return "\n".join(lines)

    # About Alish
    if any(kw in msg for kw in PORTFOLIO_KEYWORDS["about"]):
        return (
            f"👤 **About Alish Shrestha:**\n\n"
            f"Alish is an **{d['age']}-year-old** AI student and creative developer from **{d['location']}** 🇳🇵. "
            f"He is currently pursuing a **{d['degree']}** at **{d['college']}**. "
            f"Passionate about Python, Machine Learning, and building innovative web projects!"
        )

    return None


def get_system_context() -> str:
    """Returns full portfolio context for the AI system prompt."""
    d = PORTFOLIO_DATA
    return f"""
ABOUT ALISH SHRESTHA (Portfolio Owner):
- Name: {d['name']}
- Age: {d['age']} years old
- Location: {d['location']}
- Email: {d['email']}

EDUCATION:
- Currently: {d['college']} — {d['degree']}
- Secondary: {d['secondary_school']}
- Primary: {d['primary_school']}

SKILLS:
- Programming: {', '.join(d['skills']['programming'])}
- AI/ML: {', '.join(d['skills']['ai_ml'])}
- Tools: {', '.join(d['skills']['tools'])}
- Creative: {', '.join(d['skills']['creative'])}

PROJECTS:
{chr(10).join(f"- {p['name']}: {p['description']}" for p in d['projects'])}

SOCIAL:
- GitHub: {d['social']['github']}
- LinkedIn: {d['social']['linkedin']}
- Instagram: @aliisshhhhhh ({d['social']['instagram']})
- Facebook: {d['social']['facebook']}
- Discord: {d['social']['discord']}
"""
